.github/workflows/sycl_precommit.yml

```diff
@@ -27,11 +27,13 @@ jobs:
     container:
       image: ghcr.io/intel/llvm/sycl_ubuntu2004_nightly:no-drivers
     steps:
+    - name: 'PR commits + 1'
+      run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> "${GITHUB_ENV}"
     - uses: actions/checkout@v2
       with:
         ref: ${{ github.event.pull_request.head.sha }}
         persist-credentials: false
-        fetch-depth: 2
+        fetch-depth: ${{ env.PR_FETCH_DEPTH }}
```

Comment:
This approach pulls a lot of unnecessary commits. We need to pull just 2 commits: head of tested branch and merge-base with sycl branch. Before your changes we used pull request merge commit, so using fetch-depth: 2 was reasonable way to pull these two required commits.
Now as we don't use merge commit, we should probably use fetch-depth: 1, but fetch only required refs.

We have pulldowns from the community, which add a lot of commits to the sycl branch.

Reply:
Tagging @apstasen

Reply:
We only fetch commits that belong PR branch (from the point when it was branched from sycl branch). If we use fetch depth 1, then git tree used in git clang format will not work properly since sycl HEAD and PR HEAD will not have common history. PR_FETCH_DEPTH is minimum possible number of commits we need.